alibi_detect/utils/pytorch/distance.py

```diff
@@ -93,7 +115,43 @@ def batch_compute_kernel_matrix(
     return k_mat
 
 
-def mmd2_from_kernel_matrix(kernel_mat: torch.Tensor, m: int, permute: bool = False,
+def linear_mmd2(x: torch.Tensor,
+                y: torch.Tensor,
+                kernel: Callable) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    Compute maximum mean discrepancy (MMD^2) between 2 samples x and y with the
+    linear-time estimator.
+
+    Parameters
+    ----------
+    x
+        Batch of instances of shape [Nx, features].
+    y
+        Batch of instances of shape [Ny, features].
+    kernel
+        Kernel function.
+    Returns
+    -------
+    MMD^2 between the samples.
+    """
+    n = np.shape(x)[0]
+    m = np.shape(y)[0]
+    if n != m:
+        raise RuntimeError("Linear-time estimator requires equal size samples")
+    k_xx = kernel(x=x[0::2, :], y=x[1::2, :], diag=True)
+    k_yy = kernel(x=y[0::2, :], y=y[1::2, :], diag=True)
+    k_xy = kernel(x=x[0::2, :], y=y[1::2, :], diag=True)
+    k_yz = kernel(x=y[0::2, :], y=x[1::2, :], diag=True)
+
+    h = k_xx + k_yy - k_xy - k_yz
+    mmd2 = h.sum() / (n / 2.)
+    var_mmd2 = ((h * h).sum() / (n / 2.)) - (mmd2 ** 2)
```

Comment:
Should this be sample variance rather than population variance? (i.e. with 1 subtracted from (n/2.))